openshift/WFLY-14936_New_S2I_Image.adoc

```diff
@@ -0,0 +1,204 @@
+= New WildFly S2I image
+:author:           Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Today, we are relying on the WildFly s2i builder image to build a WildFly application image. 
+There is a strong coupling inside this image that implies releasing a new image for each new WildFly release. 
+This approach is not flexible. We should be able to build an application image for any WildFly release 
+(even for a locally built SNAPSHOT) without having to wait for the WildFly s2i builder image to be deployed.
+
+In addition, the WildFly server located inside the builder image, enforces a set of configuration API (a set of environment variables) 
+and bash launch scripts. At startup time, a complex CLI script generation that impacts the server startup time is mandatorily executed. 
+Having this level of configurability should be a user choice and not be enforced by the WildFly s2i image.
+
+To solve these pain points, we are removing the server and its configuration aspects from the image. 
+
+The WildFly s2i image becomes a generic image allowing to properly install and execute any provisioned WildFly server.
+ As a developer one should be able in max 2 steps to deploy to the cloud an app. Same for WildFly developers that should 
+deploy a locally built WildFly SNAPSHOT in one step (as done with WildFly Bootable JAR today). 
+
+==== Responsabilities of the WildFly s2i builder image
+
+* Provide JDK and maven tooling
```

Comment:
I think that it would be hard to achieve. We are using cct modules and s2i modules for the s2i framework. We could investigate it but my guess is that it would not come in this RFE.